### PR TITLE
Translations decaffeination

### PR DESCRIPTION
**Pull request by mserranom**
The repo contained the `.js` files emitted by `grunt` compilation. I've deleted those, and run `bulk-decaffeinate` over existing code. Last step has been to remove `grunt`, which isn't needed anymore.

### CODE REVIEW

File: app/js/translations.js [hunk 1]
```diff
@@ -1,51 +1,58 @@
...
+// TODO: This file was created by bulk-decaffeinate.
+// Sanity-check the conversion and remove this comment.
+/*
+ * decaffeinate suggestions:
+ * DS102: Remove unnecessary code created because of implicit returns
+ * DS103: Rewrite code to no longer use __guard__
+ * DS207: Consider shorter variations of null checks
+ * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
+ */
+const i18n = require("i18next");
```
Comment (das7pad) [inline app/js/translations.js]: Semicolons?

File: app/js/translations.js [hunk 2]
```diff
@@ -1,51 +1,58 @@
...
+			expressMiddlewear: i18n.handle,
+			setLangBasedOnDomainMiddlewear,
+			i18n
+		};
+	}
+};
+
+function __guard__(value, transform) {
+  return (typeof value !== 'undefined' && value !== null) ? transform(value) : undefined;
+}
```
Comment (das7pad) [inline app/js/translations.js]: Missing trailing new line?

File: package.json [hunk 3]
```diff
@@ -18,9 +20,6 @@
     "chai": "",
     "sandboxed-module": "0.2.0",
     "sinon": "",
-    "grunt-contrib-clean": "0.5.0",
-    "grunt-contrib-coffee": "0.10.0",
-    "grunt-mocha-test": "0.9.0",
```
Comment (das7pad) [inline package.json]: Let's merge https://github.com/overleaf/translations/pull/20?

**Comment by 40thieves** [comment]
While I agree that the full built scripts are unnecessary, I think we should have prettier/eslint so that when it is merged we have consistent style. WDYT?

**Comment by mserranom** [comment]
Will add eslint dependencies + scripts as in https://github.com/overleaf/logger-module/blob/master/package.json#L13-L15

**Comment by mserranom** [comment]
Added linting/formatting

File: download.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  allowedAttributes: {
+    a: ['href', 'class']
+  },
+  textFilter(text) {
+    text.replace(/\{\{/, '&#123;&#123;').replace(/\}\}/, '&#125;&#125;')
+  }
+}
+
+onesky.string.output({ platformId: '25049' }, function(err, r) {
+  if (err) {
```
Comment (mserranom) [inline download.js]: This `if (err) { }` block was caught by the linter, I assume it was an error, not a workaround for a bug in onesky.

Comment (das7pad) [reply download.js]: There was not error handing in the coffee-script version, but it's a good idea to add an error handler.

https://github.com/overleaf/translations/blob/f5de21d70a1fe60835ee7ccf7f27c8c03dbfd7f3/download.coffee#L15-L16

File: package.json [hunk 5]
```diff
@@ -6,21 +6,35 @@
     "type": "git",
     "url": "https://github.com/sharelatex/translations-sharelatex.git"
   },
+  "scripts": {
+    "lint": "node_modules/.bin/eslint .",
+    "format": "node_modules/.bin/prettier-eslint $PWD'/**/*.js' --list-different",
+    "format:fix": "node_modules/.bin/prettier-eslint $PWD'/**/*.js' --write",
+    "test": "mocha  $@ test/unit/js"
```
[the remark this answers is not in the document]
Comment (mserranom) [reply package.json]: good catch, added in https://github.com/overleaf/translations/pull/21/commits/3d354c320e829b7dbc56d4a9a8b9498ae59dccab

File: package.json [hunk 6]
```diff
@@ -32,10 +33,10 @@
...
     "eslint-plugin-promise": "^4.2.1",
     "eslint-plugin-standard": "^4.0.1",
+    "grunt-bunyan": "0.5.0",
+    "mocha": "^7.1.0",
     "prettier-eslint-cli": "^5.0.0",
     "sandboxed-module": "0.2.0",
-    "sinon": "",
-    "bunyan": "0.22.1",
-    "grunt-bunyan": "0.5.0"
+    "sinon": ""
```
Comment (das7pad) [inline package.json]: While we are at it, let's lock the sinon version.

Comment (mserranom) [reply package.json]: locked `chai` and `sinon` in https://github.com/overleaf/translations/pull/21/commits/e0e677190440a3cb44c47096902116b643a5472b

Comment (das7pad) [reply package.json]: Thanks, Miguel !

**Comment by mans0954** [comment]
Um, I think no one thought to update the Jenkins job - it's still looking for download.coffee and therefore hasn't run successfully since 16th March.

**Comment by mserranom** [comment]
Discussed separately, @mans0954 updated the Jenkins job to run the correct download script (no `Jenkinsfile` in this repo). Thanks!

**Pull request merged**